OpenSim/Actuators/Millard2012AccelerationMuscle.cpp

```diff
@@ -1256,31 +1195,20 @@ void Millard2012AccelerationMuscle::
 //==============================================================================
 // Numerical Guts: Initialization
 //==============================================================================
-SimTK::Vector Millard2012AccelerationMuscle::
-    initMuscleState(SimTK::State& s, 
-                    double aActivation, 
-                    double aSolTolerance, 
-                    int aMaxIterations,
-                    double aNewtonStepFraction) const
+Millard2012AccelerationMuscle::ResultOfInitMuscleState
+Millard2012AccelerationMuscle::initMuscleState(SimTK::Vector& result,
+                                               const SimTK::State& s,
+                                               double aActivation,
+                                               double aSolTolerance,
+                                               int aMaxIterations,
+                                               double aNewtonStepFraction) const
 {
```

Comment:
> I think it will be good to *return the result* rather than *take the result by reference and edit it*. It improves readability by separating inputs and outputs of a function. The return type could be `std::pair<ResultOfInitMuscleState,SimTK::Vector>`.

Done. The return type is now `std::pair<StatusFromInitMuscleState, ValuesFromInitMuscleState>`, where
- `StatusFromInitMuscleState` is the status flag (enum)
- `ValuesFromInitMuscleState` is a `std::map<std::string, double>` containing the calculated values